test_application_help_6 fails

Hi, I am getting the following test failure when packaging wilderness for Arch Linux:

```
......F......
======================================================================
FAIL: test_application_help_6 (tests.test_application.ApplicationTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/python-wilderness/src/wilderness-0.1.7/tests/test_application.py", line 81, in test_application_help_6
    self.assertEqual(tester.get_stdout(), self._helptext)
AssertionError: '' != 'usage: testapp [-h] [-q] command ...\n'
+ usage: testapp [-h] [-q] command ...


----------------------------------------------------------------------
Ran 13 tests in 0.005s

FAILED (failures=1)
```

Some environment info:
Python 3.10.6
Command: `python -m unittest discover`

--- a/tests/test_application.py
+++ b/tests/test_application.py
@@ -4,6 +4,7 @@ import unittest
 
 from wilderness import Application
 from wilderness.help import HelpCommand
+from wilderness.help import have_man_command
 from wilderness.tester import Tester
 
 
@@ -73,6 +74,7 @@ class ApplicationTestCase(unittest.TestCase):
         self.assertEqual(tester.get_return_code(), 1)
         self.assertEqual(tester.get_stdout(), self._helptext)
 
+    @unittest.skipUnless(have_man_command(), "no man command")
     def test_application_help_6(self):
         # Test that we get help when help command supplied
         tester = Tester(self._app)

--- a/tests/test_application.py
+++ b/tests/test_application.py
@@ -3,6 +3,7 @@
 import unittest
 
 from wilderness import Application
+from wilderness import Command
 from wilderness.help import HelpCommand
 from wilderness.help import have_man_command
 from wilderness.tester import Tester
@@ -29,10 +30,27 @@ class TestApp(Application):
         )
 
 
+class FooCommand(Command):
+    def __init__(self):
+        super().__init__(
+            "foo",
+            title="the foo command",
+            description="Description of the foo command",
+        )
+
+    def handle(self) -> int:
+        return 0
+
+
 class ApplicationTestCase(unittest.TestCase):
+    maxDiff = None
+
     def setUp(self):
         self._app = TestApp()
         self._helptext = "usage: testapp [-h] [-q] command ...\n"
+
+    def tearDown(self):
+        self._app = None
 
     def test_application_base(self):
         app = self._app
@@ -74,13 +92,31 @@ class ApplicationTestCase(unittest.TestCase):
         self.assertEqual(tester.get_return_code(), 1)
         self.assertEqual(tester.get_stdout(), self._helptext)
 
-    @unittest.skipUnless(have_man_command(), "no man command")
     def test_application_help_6(self):
         # Test that we get help when help command supplied
         tester = Tester(self._app)
         tester.test_application(["help"])
         self.assertEqual(tester.get_return_code(), 1)
         self.assertEqual(tester.get_stdout(), self._helptext)
+
+    @unittest.skipUnless(have_man_command(), "no man command")
+    def test_application_help_7(self):
+        app = self._app
+        app.add(FooCommand())
+        tester = Tester(self._app)
+        tester.test_application(["help"])
+        self.assertEqual(tester.get_return_code(), 1)
+
+        helptext = "\n".join(
+            [
+                self._helptext,
+                "Available commands:",
+                "  help  Display help information",
+                "  foo   the foo command",
+                "",
+            ]
+        )
+        self.assertEqual(tester.get_stdout(), helptext)
 
 
 if __name__ == "__main__":

--- a/wilderness/help.py
+++ b/wilderness/help.py
@@ -34,22 +34,10 @@ class HelpCommand(Command):
             description="Display help information",
         )
 
-    def _have_man(self) -> bool:
-        try:
-            cp = subprocess.run(
-                ["man", "--version"],
-                stdout=subprocess.DEVNULL,
-                stderr=subprocess.DEVNULL,
-                check=False,
-            )
-        except FileNotFoundError:
-            return False
-        return cp.returncode == 0
-
     def handle(self) -> int:
         assert self.args is not None
         cmd = self.args.command
-        if not self._have_man():
+        if not have_man_command():
             print("Error: man command not available.", file=sys.stderr)
             return 1
 
@@ -98,3 +86,16 @@ def help_action_factory(app: "Application"):
             parser.exit()
 
     return HelpAction
+
+
+def have_man_command() -> bool:
+    try:
+        cp = subprocess.run(
+            ["man", "--version"],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+            check=False,
+        )
+    except FileNotFoundError:
+        return False
+    return cp.returncode == 0

--- a/wilderness/help.py
+++ b/wilderness/help.py
@@ -36,18 +36,19 @@ class HelpCommand(Command):
 
     def handle(self) -> int:
         assert self.args is not None
-        cmd = self.args.command
-        if not have_man_command():
-            print("Error: man command not available.", file=sys.stderr)
-            return 1
-
         assert self.application
-        app_name = self.application.name
+
+        cmd = self.args.command
         if cmd is None:
             self.application.print_help()
             return 1
-        else:
-            cp = subprocess.run(["man", f"{app_name}-{cmd}"])
+
+        if not have_man_command():
+            print("Error: man command not available.", file=sys.stderr)
+            return 2
+
+        app_name = self.application.name
+        cp = subprocess.run(["man", f"{app_name}-{cmd}"])
         return cp.returncode
 
     def register(self):

